src/plugins/controls/public/options_list/components/options_list_popover.tsx

```diff
@@ -37,44 +40,70 @@ export const OptionsListPopover = ({
     typeof optionsListReducers
   >();
 
+  // Controls Services Context
+  const {
+    optionsList: { getAllowExpensiveQueries },
+  } = pluginServices.getServices();
+
   // Select current state from Redux using multiple selectors to avoid rerenders.
   const invalidSelections = select((state) => state.componentState.invalidSelections);
   const availableOptions = select((state) => state.componentState.availableOptions);
   const field = select((state) => state.componentState.field);
 
-  const hideExclude = select((state) => state.explicitInput.hideExclude);
   const hideActionBar = select((state) => state.explicitInput.hideActionBar);
+  const hideFooter = select((state) => state.explicitInput.hideFooter);
   const fieldName = select((state) => state.explicitInput.fieldName);
-  const title = select((state) => state.explicitInput.title);
   const id = select((state) => state.explicitInput.id);
 
   const [showOnlySelected, setShowOnlySelected] = useState(false);
+  const [allowExpensiveQueries, setAllowExpensiveQueries] = useState(false);
+
+  useEffect(() => {
+    const waitForAllowExpensiveQueries = async () => {
+      const allow = await getAllowExpensiveQueries();
+      setAllowExpensiveQueries(allow);
+    };
+    waitForAllowExpensiveQueries();
+  }, [getAllowExpensiveQueries]);
```

Comment:
I'm torn about this!

`getAllowExpensiveQueries` is called one time when the control is initialized - however, I need this value inside the popover so that I can decide whether or not to show the warning + cardinality badge + etc. The call to `getAllowExpensiveQueries` here **will use the cached value** so it should be relatively quick, and my goal was to avoid adding another prop to the control component...

That being said, I **could** send in `this.allowExpensiveQueries` as a prop to `OptionsListControl` as part of the embeddable render method - that would avoid having to grab this value every time the popover opens. Interested in opinions on which method would be preferred :eyes: